packages/core/src/indexing/service.test.ts

```diff
@@ -148,120 +97,163 @@ test("processEvents() calls indexing functions with correct arguments", async (c
+          startBlock: sources[1].startBlock,
+          endBlock: sources[1].endBlock,
+          maxBlockRange: sources[1].maxBlockRange,
+        },
       },
-      context: expect.objectContaining({
-        db: { TransferEvent: expect.anything(), Supply: expect.anything() },
-        network: { name: "mainnet", chainId: 1 },
-        client: expect.anything(),
-        contracts: { Erc20: expect.anything(), Pair: expect.anything() },
-      }),
-    }),
-  );
+      client: expect.any(Object),
+      db: expect.any(Object),
+    },
+  });
 
-  await service.kill();
-  await service.onIdle();
   await cleanup();
 });
 
-test("processEvent() runs setup functions before log event", async (context) => {
-  let setup = false;
-
-  const setupIndexingFunction = vi.fn(async () => {
-    setup = true;
-  });
+test("processEvent() log events", async (context) => {
+  const { common, sources, networks } = context;
+  const { syncStore, indexingStore, cleanup } = await setupDatabaseServices(
+    context,
+    { schema },
+  );
 
-  const transferIndexingFunction = vi.fn(async () => {
-    expect(setup).toBe(true);
-  });
+  const syncService = new SyncService({ common, syncStore, networks, sources });
 
   const indexingFunctions = {
     Erc20: {
-      Transfer: transferIndexingFunction,
-      setup: setupIndexingFunction,
+      Transfer: vi.fn(),
     },
   };
 
+  const indexingService = createIndexingService({
+    indexingFunctions,
+    common,
+    sources,
+    networks,
+    syncService,
+    indexingStore,
+    schema,
+  });
+
+  const rawEvents = await getEventsErc20(sources);
+  const events = decodeEvents(indexingService, rawEvents);
+  const result = await processEvents(indexingService, {
+    events,
+  });
+  expect(result).toStrictEqual({ status: "success" });
+
+  expect(indexingFunctions.Erc20.Transfer).toHaveBeenCalledTimes(2);
+  expect(indexingFunctions.Erc20.Transfer).toHaveBeenCalledWith({
+    event: {
+      name: "Transfer",
+      args: expect.any(Object),
+      log: expect.any(Object),
+      block: expect.any(Object),
+      transaction: expect.any(Object),
+    },
+    context: {
+      network: { chainId: 1, name: "mainnet" },
+      contracts: {
+        Erc20: {
+          abi: expect.any(Object),
+          address: checksumAddress(sources[0].criteria.address as Address),
+          startBlock: sources[0].startBlock,
+          endBlock: sources[0].endBlock,
+          maxBlockRange: sources[0].maxBlockRange,
+        },
+        Pair: {
+          abi: expect.any(Object),
+          address: checksumAddress(sources[1].criteria.address as Address),
+          startBlock: sources[1].startBlock,
+          endBlock: sources[1].endBlock,
+          maxBlockRange: sources[1].maxBlockRange,
+        },
+      },
+      client: expect.any(Object),
+      db: expect.any(Object),
+    },
+  });
+
+  await cleanup();
+});
+
+test("processEvents killed", async (context) => {
```

Comment:
`"processEvents does not process events and returns killed status while killed"`